src/util/union_find.cpp

```diff
@@ -80,13 +80,13 @@ void unsigned_union_find::re_root(size_type old_root, size_type new_root)
   old_root=find(old_root);
 
   // same set?
-  // assert(find(new_root)==old_root);
+  // assert(find(new_root)==old_root); // Commented out dead code? To review
   if(find(new_root)!=old_root)
     return;
 
   // make sure we actually do something
-  assert(new_root!=old_root);
-  assert(nodes[old_root].count>=2);
+  INVARIANT(new_root!=old_root, "Ensure the algorithm roots differ"); // TODO review
```

Comment:
Previous conditions make sure that `old_root` is actually a root and that the root of `new_root` isn't `old_root`, therefore `new_root` is in a different tree compared to `old_root` and therefore also a different node. (bit of a mouthful)